test/view_models/advanced_viewmodel_test.dart

```diff
@@ -0,0 +1,143 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:hng/ui/view/advanced/advanced_viewmodel.dart';
+import 'package:hng/utilities/enums.dart';
+import 'package:mockito/mockito.dart';
+import '../helpers/test_helpers.dart';
+
+AdvancedViewModel _getModel() => AdvancedViewModel();
+
+void main() {
+  group('AdvancedViewmodelTest -', () {
+    setUp(() => registerServices());
+    tearDown(() => unregisterServices());
+
+    group('changeSkinTone -', () {
+      test('when called, call dialog service to show custom dialog', () async {
+        final dialogService = getAndRegisterDialogServiceMock();
+        var model = _getModel();
+        await model.changeSkinTone();
+        verify(dialogService.showCustomDialog(variant: DialogType.skinTone));
+      });
+    });
+
+    group('toggleShowImagePreviews -', () {
+      test(
+          'When toggleShowImagePreviews is called should notifyListeners to rebuild UI',
+          () {
+        var model = _getModel();
+        bool called = false;
+        model.addListener(() {
+          called = true;
+        });
+        model.toggleShowImagePreviews(true);
+
+        expect(called, true);
+      });
+      test(
+          'When toggleShowImagePreviews is called and value is true, showImagePreviews should be true',
+          () {
+        var model = _getModel();
+        model.toggleShowImagePreviews(true);
+        expect(model.showImagePreviews, true);
+      });
+
+      test(
+          'When toggleShowImagePreviews is called and value is false, showImagePreviews should be false',
+          () {
+        var model = _getModel();
+        model.toggleShowImagePreviews(false);
+        expect(model.showImagePreviews, false);
+      });
+    });
+
+    group('toggleDisplayTypingIndicators -', () {
+      test(
+          'When toggleDisplayTypingIndicators is called should notifyListeners to rebuild UI',
+          () {
+        var model = _getModel();
+        bool called = false;
+        model.addListener(() {
```

Comment:
why is this here
if ```model.toggleDisplayTypinIndicators(true);``` is here

Reply:
It's checking that the listener rebuilds the UI. I don't quite get your question. 
@pauleke65

Reply:
i meant your assigning true to called in the add listener
and also passing it through the display typing indicators.
so which is actually changing the value